buffer/src/main/java/io/netty/buffer/api/BufferAllocator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import io.netty.buffer.api.pool.PooledBufferAllocator;
+
+import java.nio.ByteOrder;
+import java.util.function.Supplier;
+
+/**
+ * Interface for allocating {@link Buffer}s.
+ */
+public interface BufferAllocator extends AutoCloseable {
```

Comment:
Is this intended to work with try-with-resources?     I can't imagine a use where that would be needed.    Also, it makes it kind of wierd because now the question of who should close it comes up.   In my own uses of Netty4's allocator, I typically pass it around to classes and stuff it in a final variable. (or pull it out of the channel).   I would imagine few users would need to or feel comfortable closing it.  

One idea: consider splitting the lifecycle part of the buffer allocator into a subclass.   Existing uses of BufferAllocator would still be correct, but lower level code could use the more specific type.  (CloseableBufferAllocator or ManagedBufferAllocator come to midnd).

Reply:
Splitting the life cycle would be inconvenient for tests. I don't think this will be a big deal in practice, so I'd like to wait and see how things look once the new API is more integrated into Netty.